webapp/js/components/Utils.js

```diff
@@ -23,3 +23,35 @@ export function T(message) {
     return msg
 }
 
+export function isHTTPError(response) {
+    // Check for an HTTP error
+    if ((response.statusCode >= 300) && (response.statusCode != 400)) {
```

Comment:
The API returns an error 400 for "expected errors" i.e. ones that have been caught and returned as a JSON response. So, the normal responses will always be 200 or 400. Anything else is something unexpected happening at the server and it is likely that the response won't be JSON. So this is to handle the unexpected errors gracefully.